File: output/code.cs
using System.Collections.Generic;
using System;

public class Library {
    public string Name { get; set; }
    public string Address { get; set; }
    public List<Book> Books { get; set; }

    public Library(string name, string address)
    {
        this.Name = name;
        this.Address = address;
        this.Books = new List<Book>();
    }

    public void AddBook(Book book)
    {
        if (!Books.Contains(book))
        {
            Books.Add(book);
            book.Library = this;
        }
    }

    public override string ToString()
    {
        return "Library{Name=" + Name + ", Address=" + Address + ", Books=" + Books.Count + "}";
    }
}
public class Book {
    public DateTime Release { get; set; }
    public string Title { get; set; }
    public int Pages { get; set; }
    public Library Library { get; set; }
    public List<Author> Authors { get; set; }

    public Book(DateTime release, string title, int pages)
    {
        this.Release = release;
        this.Title = title;
        this.Pages = pages;
        this.Library = null;
        this.Authors = new List<Author>();
    }

    public void AddAuthor(Author author)
    {
        if (!Authors.Contains(author))
        {
            Authors.Add(author);
            author.AddBook(this);
        }
    }

    public override string ToString()
    {
        return "Book{Release=" + Release + ", Title=" + Title + ", Pages=" + Pages + ", Library=" + (Library != null ? Library.Name : "None") +  ", Authors=" + Authors.Count + "}";
    }
}
public class Author {
    public string Email { get; set; }
    public string Name { get; set; }
    public List<Book> Books { get; set; }

    public Author(string email, string name)
    {
        this.Email = email;
        this.Name = name;
        this.Books = new List<Book>();
    }

    public void AddBook(Book book)
    {
        if (!Books.Contains(book))
        {
            Books.Add(book);
            book.AddAuthor(this);
        }
    }

    public override string ToString()
    {
        return "Author{Email=" + Email + ", Name=" + Name + ", Books=" + Books.Count + "}";
    }
}

File: output/code.java
import java.util.ArrayList;
import java.util.List;
import java.time.LocalDateTime;

public class Library {
    private String name; 
    private String address; 
    private List<Book> books;

    public Library(String name, String address){
        this.name = name;
        this.address = address;
        this.books = new ArrayList<>();
    }    

    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public String getAddress() {
        return address;
    }
    public void setAddress(String address) {
        this.address = address;
    }
    public List<Book> getBooks() {
        return books;
    }

    public void addBook(Book book) {
        if (!books.contains(book)) {
            books.add(book);
            book.setLibrary(this);
        }
    }

    @Override
    public String toString() {
        return "Library{name=" + name + ", address=" + address + ", books=" + books.size() + "}";
    }
}

public class Book {
    private LocalDateTime release; 
    private String title; 
    private int pages; 
    private Library library;
    private List<Author> authors;

    public Book(LocalDateTime release, String title, int pages){
        this.release = release;
        this.title = title;
        this.pages = pages;
        this.library = null; 
        this.authors = new ArrayList<>();
    }    

    public LocalDateTime getRelease() {
        return release;
    }
    public void setRelease(LocalDateTime release) {
        this.release = release;
    }
    public String getTitle() {
        return title;
    }
    public void setTitle(String title) {
        this.title = title;
    }
    public int getPages() {
        return pages;
    }
    public void setPages(int pages) {
        this.pages = pages;
    }
    public Library getLibrary() {
        return library;
    }

    public void setLibrary(Library library) {
        if (this.library != library) {
            this.library = library;
            library.addBook(this);
        }
    }
    public List<Author> getAuthors() {
        return authors;
    }
    
    public void addAuthor(Author author) {
        if (!authors.contains(author)) {
            authors.add(author);
            author.addBook(this);
        }
    }

    @Override
    public String toString() {
        return "Book{release=" + release + ", title=" + title + ", pages=" + pages + ", library=" + (library != null ? library.getName() : "None") +  ", authors=" + authors.size() + "}";
    }
}

public class Author {
    private String email; 
    private String name; 
    private List<Book> books;

    public Author(String email, String name){
        this.email = email;
        this.name = name;
        this.books = new ArrayList<>();
    }    

    public String getEmail() {
        return email;
    }
    public void setEmail(String email) {
        this.email = email;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public List<Book> getBooks() {
        return books;
    }

    public void addBook(Book book) {
        if (!books.contains(book)) {
            books.add(book);
            book.addAuthor(this);
        }
    }

    @Override
    public String toString() {
        return "Author{email=" + email + ", name=" + name + ", books=" + books.size() + "}";
    }
}

